scipy/stats/_mannwhitneyu.py

```diff
@@ -0,0 +1,367 @@
+
+    # Continuity correction. I still need to find a reference for this.
+    if continuity:
+        numerator -= 0.5 * (U != mu)
+
+    z = numerator / s
+    return z
+
+
+def _mwu_input_validation(x, y, use_continuity, alternative, axis, method):
+    ''' Input validation and standardization for mannwhitneyu2 '''
+    # Would use np.asarray_chkfinite, but infs are OK
+    x, y = np.atleast_1d(x), np.atleast_1d(y)
+    if np.isnan(x).any() or np.isnan(y).any():
+        raise ValueError('`x` and `y` must not contain NaNs.')
+    if np.size(x) == 0 or np.size(y) == 0:
+        raise ValueError('`x` and `y` must be of nonzero size.')
+
+    bools = {True, False}
+    if use_continuity not in bools:
+        raise ValueError(f'`use_continuity` must be one of {bools}.')
+
+    alternatives = {"two-sided", "less", "greater"}
+    alternative = alternative.lower()
+    if alternative not in alternatives:
+        raise ValueError(f'`alternative` must be one of {alternatives}.')
+
+    axis_int = int(axis)
+    if axis != axis_int:
+        raise ValueError('`axis` must be an integer.')
+
+    methods = {"asymptotic", "exact"}
+    method = method.lower()
+    if method not in methods:
+        raise ValueError(f'`method` must be one of {methods}.')
+
+    return x, y, use_continuity, alternative, axis_int, method
+
+
+mwu_result = make_dataclass("MannWhitneyUResult", ("statistic", "pvalue"))
+# Using `nametuple` for now to pass existing mannwhitneyu tests without
+# modification
+MannwhitneyuResult = namedtuple('MannwhitneyuResult', ('statistic', 'pvalue'))
+
+
+def mannwhitneyu2(x, y, use_continuity=True, alternative="two-sided",
+                  axis=0, method="asymptotic"):
+    r'''Perform the Mann-Whitney U rank test on two independent samples.
+
+    The Mann-Whitney U test is a nonparametric test of the null hypothesis
+    that the underlying distribution of sample `x` is neither stochastically
+    greater nor stochastically less than the uderlying distribution of sample
+    `y`.
+
+    Parameters
+    ----------
+    x, y : array-like
+        N-d arrays of samples. The arrays must be broadcastable except along
+        the dimension given by `axis`.
+    use_continuity : bool, optional
+            Whether a continuity correction (1/2) should be applied.
+            Default is True when `method` is ``'asymptotic'``; has no effect
+            otherwise.
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis.
+        The following options are available:
+
+        * 'two-sided' (default)
+        * 'less': one-sided
+        * 'greater': one-sided
+
+    axis : int, optional
+        Axis along which to perform the test. Default is 0.
+    method : {'asymptotic', 'exact'}, optional
+        Selects the method used to calculate the *p*-value. The following
+        options are available.
+
+        * ``'asymptotic'``: (default) compares the standardized test statistic
```

Comment:
are there different ways to correct for ties as in the Wilcoxon test?

Reply:
Hmm I will look into it. Probably - I ran across a paper while reviewing gh-13438 that might have a way to deal with ties even in the exact case. But since it is not a feature of the original `mannwhitneyu` nor was it one of the requested features, I might ask to leave it for a followup PR unless it has implications for the interface.

Reply:
my question was out of curiosity since for the Wilcoxcon test, there are different approaches. in any case, it could be added later